### ISSUE TITLE
Escaping

### ISSUE THREAD

**Issue by drtconway** [PMCC-BioinformaticsCore/janis-core#11]
Hi Janis,

I think the tool is actually irrelevant, but I'm attaching it anyway. (oh. I can't attach Python? I'm pasting the code down the bottom.)

When I run:

```
$ janis run -o wibble regex_search.py --expression '([^/]*)[.]fastq([.]gz)?' --replacement '\1' --text '/foo/bar/baz.fastq.gz’
```

Execution fails, and the following can be seen in the engine.log file:

```
    python RegexSearch-script.py \
      --expression ([^/]*)[.]fastq([.]gz)? \
      --replacement \1 \
      --text /foo/bar/baz.fastq.gz`
```

It's obvious why this fails - the arguments are not being escaped.

```python
from typing import List

from janis_core import String, TOutput
from janis_bioinformatics.tools.bioinformaticstoolbase import BioinformaticsPythonTool

class RegexSearch(BioinformaticsPythonTool):
    @staticmethod
    def code_block(expression: String, replacement: String, text: String):
        import re
        m = re.search(expression, text)
        if m is None:
            raise Exception(
                f"No match for the expression '{expression}' in the text '{text}'"
            )
        res = m.expand(replacement)
        return {'result': res}

    def outputs(self) -> List[TOutput]:
        return [TOutput("result", String)]

    def id(self) -> str:
        return "RegexSearch"

    def friendly_name(self):
        return "Apply a regular expression template"

    def version(self):
        return "v0.1.0"

    def tool_provider(self):
        return "Peter MacCallum Cancer Centre"

    def bind_metadata(self):
        self.metadata.documentation = (
            "repack the filenames from splitting fastq files."
        )
        self.metadata.creator = "Thomas Conway"
        self.metadata.dateCreated = "2020-03-25"
        self.metadata.version = "0.1.0"
```

**Comment by illusional** [comment]
After a little investigation, this is actually a problem with the WDL command template generation: I'll tag this issue (https://github.com/PMCC-BioinformaticsCore/python-wdlgen/issues/2) because I intend to fix this soon.

**Comment by illusional** [comment]
Hey @drtconway, later update to this. I've redone the parameter quoting in the the [Expressions PR](https://github.com/PMCC-BioinformaticsCore/janis-core/pull/10) (https://github.com/PMCC-BioinformaticsCore/janis-core/pull/10).

I'm still waiting on a few related PRs before I can release that set of features, but it's coming!

**Comment by illusional** [comment]
This has been released in `v0.10.x`.
